### ISSUE TITLE
[BIG BAD ISSUE] Remove Dependency on Both Credentials Being Present/Valid

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/cloudware#23]
Remove the data from the Azure keys in `cloudware.yml` and the following will be seen:
```
[vagrant@controller cloudware]$ cloudware machine list --domain aws
error: {ing available machines
  "message": "Couldn't login to Azure, please verify your tenant id, client id and client secret",
  "request": null,
  "response": null
}. Use --trace to view backtrace
[vagrant@controller cloudware]$ cloudware machine list --domain aws --trace
/opt/cloudware/vendor/ruby/2.4.0/gems/ms_rest_azure-0.9.0/lib/ms_rest_azure/credentials/application_token_provider.rb:108:in `acquire_token': { (MsRestAzure::AzureOperationError)
  "message": "Couldn't login to Azure, please verify your tenant id, client id and client secret",
  "request": null,
  "response": null
}
	from /opt/cloudware/vendor/ruby/2.4.0/gems/ms_rest_azure-0.9.0/lib/ms_rest_azure/credentials/application_token_provider.rb:69:in `get_authentication_header'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/ms_rest-0.7.2/lib/ms_rest/credentials/token_credentials.rb:49:in `sign_request'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/ms_rest-0.7.2/lib/ms_rest/service_client.rb:51:in `block in make_request_async'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/ms_rest-0.7.2/lib/ms_rest/http_operation_request.rb:82:in `block (2 levels) in run_promise'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/connection.rb:384:in `block in run_request'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/connection.rb:398:in `block in build_request'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/request.rb:26:in `block in create'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/request.rb:25:in `tap'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/request.rb:25:in `create'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/connection.rb:394:in `build_request'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/faraday-0.13.1/lib/faraday/connection.rb:379:in `run_request'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/ms_rest-0.7.2/lib/ms_rest/http_operation_request.rb:80:in `block in run_promise'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/safe_task_executor.rb:24:in `block in execute'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `block in synchronize'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `synchronize'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `synchronize'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/safe_task_executor.rb:19:in `execute'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/promise.rb:531:in `block in realize'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
```

If the Azure details are removed _completely_ from the yaml file then the following occurs:
```
[vagrant@controller cloudware]$ cloudware machine list --domain aws
error: undefined method `[]' for nil:NilClass. Use --trace to view backtrace
[vagrant@controller cloudware]$ cloudware machine list --domain aws --trace
/opt/cloudware/lib/cloudware/config.rb:36:in `initialize': undefined method `[]' for nil:NilClass (NoMethodError)
	from /opt/cloudware/lib/cloudware.rb:34:in `new'
	from /opt/cloudware/lib/cloudware.rb:34:in `config'
	from /opt/cloudware/lib/cloudware/aws.rb:47:in `config'
	from /opt/cloudware/lib/cloudware/aws.rb:65:in `valid_credentials?'
	from /opt/cloudware/lib/cloudware/aws.rb:51:in `load_config'
	from /opt/cloudware/lib/cloudware/aws.rb:43:in `initialize'
	from /opt/cloudware/lib/cloudware/machine.rb:99:in `new'
	from /opt/cloudware/lib/cloudware/machine.rb:99:in `list'
	from /opt/cloudware/lib/cloudware/cli.rb:221:in `block (2 levels) in <class:CLI>'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/command.rb:153:in `run'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/runner.rb:446:in `run_active_command'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/runner.rb:68:in `run!'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/delegates.rb:15:in `run!'
	from /opt/cloudware/bin/cloudware:32:in `<main>'
```

Similar is seen when the AWS details are empty
```
[vagrant@controller cloudware]$ cloudware machine list --domain azure
error: Authorization header or parameters are not formatted correctly.. Use --trace to view backtrace
[vagrant@controller cloudware]$ cloudware machine list --domain azure --trace
/opt/cloudware/lib/cloudware/cli.rb:237:in `rescue in block (2 levels) in <class:CLI>': Authorization header or parameters are not formatted correctly. (RuntimeError)
	from /opt/cloudware/lib/cloudware/cli.rb:217:in `block (2 levels) in <class:CLI>'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/command.rb:153:in `run'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/runner.rb:446:in `run_active_command'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/runner.rb:68:in `run!'
	from /opt/cloudware/vendor/ruby/2.4.0/gems/commander-4.4.3/lib/commander/delegates.rb:15:in `run!'
	from /opt/cloudware/bin/cloudware:32:in `<main>'
```

And when the AWS is removed it's identical to the Azure details being removed.

**Comment by ColonelPanicks** [comment]
Partially fixed in 42c94a1

**Comment by ColonelPanicks** [comment]
This has been addressed via 42c94a1 and 3f40f5d
